File: src/main/java/fr/traqueur/energylib/commands/EnergyCommand.java
package fr.traqueur.energylib.commands;

import fr.traqueur.commands.api.Arguments;
import fr.traqueur.commands.api.Command;
import fr.traqueur.energylib.EnergyLib;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class EnergyCommand extends Command<EnergyLib> {

    public EnergyCommand(EnergyLib plugin) {
        super(plugin, "energy-admin");

        this.setDescription("The admin command of the plugin energylib.");
        this.setPermission("energy.admin");
        this.setUsage("/energy-admin");

        this.setGameOnly(true);
    }

    @Override
    public void execute(CommandSender commandSender, Arguments arguments) {
        Player player = (Player) commandSender;
        player.sendMessage("§cThis command is not implemented yet.");
    }
}

File: src/main/java/fr/traqueur/energylib/commands/EnergyCommand.kt
package fr.traqueur.energylib.commands

import fr.traqueur.commands.api.Arguments
import fr.traqueur.commands.api.Command
import fr.traqueur.energylib.EnergyLib
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class EnergyCommand(plugin: EnergyLib?) : Command<EnergyLib?>(plugin, "energy-admin") {
    init {
        this.setDescription("The admin command of the plugin energylib.")
        this.setPermission("energy.admin")
        this.setUsage("/energy-admin")

        this.addSubCommand(
            DeleteCommand(plugin!!),
            ListCommand(plugin),
            ShowCommand(plugin)
        )

        this.setGameOnly(true)
    }

    override fun execute(commandSender: CommandSender?, arguments: Arguments?) {
        val player = commandSender as Player
        player.sendMessage("§cUsage: /energy-admin <delete/list/show>")
    }
}
